src/main.rs

```diff
@@ -505,17 +510,41 @@ fn cp_r(src: &Path,
                 }
             }
             _ => ()
+        };
+
+        // Join pathname components individually to make sure that the joined
+        // path uses the correct directory separators everywhere, since
+        // `relative` may use Unix-style and `dst` may require Windows-style
+        // backslashes.
+        let dst = relative.iter().fold(dst.to_owned(), |acc, component| {
+            acc.join(&component)
+        });
+
+        // Create any necessary directories for the given file.  We may need
+        // to create multiple directories, so loop until we find a directory
+        // we've already created.
+        let mut to_create = Vec::new();
```

Comment:
Instead of this loop, could `create_dir_all` be used perhaps?

Reply:
I'd be fine with just doing `create_dir_all` and dispensing with the loop and `created_directories` entirely.  The intent here was to try to minimize the amount of filesystem I/O, but maybe it doesn't matter that much?  It felt like lots of deeply nested directories, possibly combined with numerous files in them, would make `create_dir_all` expensive, but that might just be premature optimization.